torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -101,6 +103,179 @@ std::unique_ptr<RpcCommandBase> deserializeResponse(const Message& response) {
   }
 }
 
+namespace {
+
+// Helper for wireDeserialize() below.
+//
+// The format we use below looks like:
+//    section_name_1 size_1\n
+//    section_name_2 size_2\n
+//    ..
+//    \n
+//    [sections in order]
+//
+// Sections themselves include:
+//    - "payload" - the payload bits
+//    - "meta"    - metadata for the unpickler
+//    - "0" ...   - tensor sections for the unpickler
+//
+// Note that per the header comments, the format is subject to change,
+// and is best used for rpcs, rather than persistent disk storage.
+std::unordered_map<std::string, std::pair<const char*, size_t>>
+parseWireSections(const void* data, size_t data_size) {
+  const char* ptr = static_cast<const char*>(data);
+  const char* endp = ptr + data_size;
+
+  std::vector<std::pair<std::string, size_t>> header_ents;
+  bool ok = false;
+  while (ptr != endp) {
+    if (*ptr == '\n') {
+      ok = true;
+      break;
+    }
+    const char* num_ptr = ptr;
+    while (*ptr != ' ' && ptr != endp) {
+      ptr++;
+    }
+    if (ptr == endp) {
+      break;
+    }
+    std::string name(num_ptr, ptr - num_ptr);
+    if (++ptr == endp) {
+      break; // past the ' '
+    }
+    const char* size_ptr = ptr;
+    while (*ptr != '\n' && ptr != endp) {
+      ptr++;
+    }
+    if (ptr == endp) {
+      break;
+    }
+    size_t sz = c10::stoll(std::string(size_ptr, ptr - size_ptr));
+    header_ents.emplace_back(std::make_pair(name, sz));
+    ++ptr; // past the '\n'
+  }
+  ++ptr; // past the final \n
+
```

Comment:
Changed to a separate pre-check for ok here.

Agree this makes the code more clear. I was simply trying to avoid an extra throw statement, which was probably slightly silly.